src/Mod/Sketcher/Gui/TaskSketcherTool.cpp

```diff
@@ -0,0 +1,911 @@
+    }
+    else setParameterFocus(3);
+
+    if (isWidgetActive) {
+        sketchView->mouseMove(sketchView->prvMoveCursorPos, sketchView->prvMoveViewer);
+    }
+}
+void SketcherToolWidget::emitSetparameterFour(double val)
+{
+    isSettingSet[3] = 1;
+    toolParameters[3] = val;
+    if (unitChangeTriggeredValueChanged) {
+        unitChangeTriggeredValueChanged = 0;
+    }
+    else setParameterFocus(4);
+
+    if (isWidgetActive) {
+        sketchView->mouseMove(sketchView->prvMoveCursorPos, sketchView->prvMoveViewer);
+    }
+}
+void SketcherToolWidget::emitSetparameterFive(double val)
+{
+    isSettingSet[4] = 1;
+    toolParameters[4] = val;
+    if (unitChangeTriggeredValueChanged) {
+        unitChangeTriggeredValueChanged = 0;
+    }
+    else setParameterFocus(5);
+
+    if (isWidgetActive) {
+        sketchView->mouseMove(sketchView->prvMoveCursorPos, sketchView->prvMoveViewer);
+    }
+}
+void SketcherToolWidget::emitSetparameterSix(double val)
+{
+    isSettingSet[5] = 1;
+    toolParameters[5] = val;
+
+    if (isWidgetActive) {
+        sketchView->mouseMove(sketchView->prvMoveCursorPos, sketchView->prvMoveViewer);
+    }
+}
+
+bool SketcherToolWidget::isCheckBoxChecked(int i) {
+    if (i == 1) {
+        return ui->checkBoxPreserveFilletChamferCorner->isChecked();
+    }
+    else if (i == 2) {
+        return ui->checkBoxTS2->isChecked();
+    }
+    return 0;
+}
+
+void SketcherToolWidget::setSettings(int toolSelected)
+{
+    //reset and hide all settings (case 0)
+    isWidgetActive = 0;
+    toolParameters.clear();
+    isSettingSet.clear();
+
+    ui->label->setVisible(0);
+    ui->label2->setVisible(0);
+    ui->label3->setVisible(0);
+    ui->label4->setVisible(0);
+    ui->label5->setVisible(0);
+    ui->label6->setVisible(0);
+    ui->notice->setVisible(0);
+    setparameter(0, 0);
+    setparameter(0, 1);
+    setparameter(0, 2);
+    setparameter(0, 3);
+    setparameter(0, 4);
+    setparameter(0, 5);
+    setUnit(Base::Unit::Length, 0);
+    setUnit(Base::Unit::Length, 1);
+    setUnit(Base::Unit::Length, 2);
+    setUnit(Base::Unit::Length, 3);
+    setUnit(Base::Unit::Length, 4);
+    setUnit(Base::Unit::Length, 5);
+    unitChangeTriggeredValueChanged = 0;
+
+    ui->parameterOne->setVisible(0);
+    ui->parameterTwo->setVisible(0);
+    ui->parameterThree->setVisible(0);
+    ui->parameterFour->setVisible(0);
+    ui->parameterFive->setVisible(0);
+    ui->parameterSix->setVisible(0);
+    ui->checkBoxPreserveFilletChamferCorner->setVisible(0);
+    ui->checkBoxTS2->setVisible(0);
+    //sketchView->toolSettings->hideGroupBox();
+
+    //Give the focus back to the viewproviderSketcher
+    //Gui::MDIView* mdi = Gui::Application::Instance->activeDocument()->getActiveView();
+    //mdi->setFocus();
+    QMdiArea* mdi = qobject_cast<QMdiArea*>(Gui::MainWindow::getInstance()->centralWidget());
+    if (!mdi) return;
+    mdi->activeSubWindow()->widget()->setFocus();
+
+    switch (toolSelected) {

```

Comment:
This kind of solution is not very flexible for changes. For example, there is a checkbox signal exclusively for chamfer.

Ideally, this widget would have generic code only and the specific code would be provided by the specific DrawHandler (or a specific class associated to the DrawHandler).